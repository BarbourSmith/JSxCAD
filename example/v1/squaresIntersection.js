import { difference, intersection, square, writePdf } from '@jsxcad/api-v1';

export const main = async () =>
  await writePdf({ path: 'tmp/squaresIntersection.pdf' },
                 intersection(difference(square(10), square(9)).translate([-2, -2]),
                              difference(square(10), square(9)).translate([2, 2])));
