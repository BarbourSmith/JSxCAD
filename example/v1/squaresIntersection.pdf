%PDF-1.5
1 0 obj << /Pages 2 0 R /Type /Catalog >> endobj
2 0 obj << /Count 1 /Kids [ 3 0 R ] /Type /Pages >> endobj
3 0 obj <<
  /Contents 4 0 R
  /MediaBox [ 0 0 595.275590176 841.889763249 ]
  /Parent 2 0 R
  /Type /Page
>>
endobj
4 0 obj << >>
stream
0.096000000 w
0.000000000 840.472440415 m
1.417322834 840.472440415 l
1.417322834 841.889763249 l
0.000000000 841.889763249 l
h
S
15.590551171 824.881889244 m
17.007874005 824.881889244 l
17.007874005 826.299212078 l
15.590551171 826.299212078 l
h
S
endstream
endobj
trailer << /Root 1 0 R /Size 4 >>
%%EOF