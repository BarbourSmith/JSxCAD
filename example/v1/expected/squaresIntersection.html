<html>
  <head>
    <title>JSxCAD Viewer</title>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0"
    />
  </head>
  <body>
    <script type="module">
      import { toDomElement } from 'https://gitcdn.link/cdn/jsxcad/JSxCAD/master/es6/jsxcad-ui-notebook.js';

      const notebook = [
        {
          view: {
            width: 512,
            height: 256,
            position: [100, -100, 100],
            geometry: {
              type: 'layout',
              layout: {
                size: [8.000000000000002, 8.000000000000002],
                margin: 1,
                title: '8 x 8 : ',
              },
              marks: [
                [0, 0, 0],
                [8.000000000000002, 8.000000000000002, 0],
              ],
              content: [
                {
                  type: 'item',
                  content: [
                    {
                      type: 'disjointAssembly',
                      content: [
                        {
                          type: 'disjointAssembly',
                          content: [
                            {
                              type: 'z0Surface',
                              z0Surface: [
                                [
                                  [3.0000000000000018, -2.500000000000001, 0],
                                  [2.5000000000000018, -2.500000000000001, 0],
                                  [2.5000000000000018, -3.000000000000001, 0],
                                  [3.0000000000000018, -3.000000000000001, 0],
                                ],
                                [
                                  [-3, 2.5, 0],
                                  [-2.4999999999999982, 2.5, 0],
                                  [-2.499999999999999, 3, 0],
                                  [-3, 3, 0],
                                ],
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                {
                  type: 'disjointAssembly',
                  content: [
                    {
                      type: 'disjointAssembly',
                      content: [
                        {
                          type: 'disjointAssembly',
                          content: [
                            {
                              type: 'paths',
                              tags: ['compose/non-positive', 'color/red'],
                              paths: [
                                [
                                  [4.000000000000001, 4.000000000000001, 0],
                                  [-4.000000000000001, 4.000000000000001, 0],
                                  [-4.000000000000002, -4, 0],
                                  [4, -4.000000000000002, 0],
                                ],
                              ],
                            },
                          ],
                        },
                        {
                          type: 'paths',
                          tags: ['compose/non-positive', 'color/red'],
                          paths: [
                            [
                              null,
                              [-3.857142857142858, 4.625000000000001, 0],
                              [-3.9107142857142865, 4.607142857142858, 0],
                              [-3.9285714285714293, 4.571428571428572, 0],
                              [-3.9285714285714293, 4.5357142857142865, 0],
                              [-3.9107142857142865, 4.500000000000001, 0],
                              [-3.875000000000001, 4.482142857142858, 0],
                              [-3.8035714285714293, 4.464285714285715, 0],
                              [-3.750000000000001, 4.446428571428572, 0],
                              [-3.7142857142857153, 4.4107142857142865, 0],
                              [-3.696428571428572, 4.375000000000001, 0],
                              [-3.696428571428572, 4.321428571428572, 0],
                              [-3.7142857142857153, 4.2857142857142865, 0],
                              [-3.732142857142858, 4.267857142857144, 0],
                              [-3.7857142857142865, 4.250000000000001, 0],
                              [-3.857142857142858, 4.250000000000001, 0],
                              [-3.9107142857142865, 4.267857142857144, 0],
                              [-3.9285714285714293, 4.2857142857142865, 0],
                              [-3.9464285714285725, 4.321428571428572, 0],
                              [-3.9464285714285725, 4.375000000000001, 0],
                              [-3.9285714285714293, 4.4107142857142865, 0],
                              [-3.8928571428571437, 4.446428571428572, 0],
                              [-3.8392857142857153, 4.464285714285715, 0],
                              [-3.7678571428571437, 4.482142857142858, 0],
                              [-3.732142857142858, 4.500000000000001, 0],
                              [-3.7142857142857153, 4.5357142857142865, 0],
                              [-3.7142857142857153, 4.571428571428572, 0],
                              [-3.732142857142858, 4.607142857142858, 0],
                              [-3.7857142857142865, 4.625000000000001, 0],
                              [-3.857142857142858, 4.625000000000001, 0],
                            ],
                            [null],
                            [null],
                            [
                              null,
                              [-3.3035714285714293, 4.500000000000001, 0],
                              [-3.1071428571428577, 4.250000000000001, 0],
                            ],
                            [
                              null,
                              [-3.1071428571428577, 4.500000000000001, 0],
                              [-3.3035714285714293, 4.250000000000001, 0],
                            ],
                            [null],
                            [null],
                            [
                              null,
                              [-2.625000000000001, 4.625000000000001, 0],
                              [-2.6785714285714293, 4.607142857142858, 0],
                              [-2.696428571428572, 4.571428571428572, 0],
                              [-2.696428571428572, 4.5357142857142865, 0],
                              [-2.6785714285714293, 4.500000000000001, 0],
                              [-2.6428571428571432, 4.482142857142858, 0],
                              [-2.571428571428572, 4.464285714285715, 0],
                              [-2.5178571428571432, 4.446428571428572, 0],
                              [-2.4821428571428577, 4.4107142857142865, 0],
                              [-2.464285714285715, 4.375000000000001, 0],
                              [-2.464285714285715, 4.321428571428572, 0],
                              [-2.4821428571428577, 4.2857142857142865, 0],
                              [-2.500000000000001, 4.267857142857144, 0],
                              [-2.5535714285714293, 4.250000000000001, 0],
                              [-2.625000000000001, 4.250000000000001, 0],
                              [-2.6785714285714293, 4.267857142857144, 0],
                              [-2.696428571428572, 4.2857142857142865, 0],
                              [-2.7142857142857153, 4.321428571428572, 0],
                              [-2.7142857142857153, 4.375000000000001, 0],
                              [-2.696428571428572, 4.4107142857142865, 0],
                              [-2.6607142857142865, 4.446428571428572, 0],
                              [-2.6071428571428577, 4.464285714285715, 0],
                              [-2.5357142857142865, 4.482142857142858, 0],
                              [-2.500000000000001, 4.500000000000001, 0],
                              [-2.4821428571428577, 4.5357142857142865, 0],
                              [-2.4821428571428577, 4.571428571428572, 0],
                              [-2.500000000000001, 4.607142857142858, 0],
                              [-2.5535714285714293, 4.625000000000001, 0],
                              [-2.625000000000001, 4.625000000000001, 0],
                            ],
                            [null],
                            [null],
                            [
                              null,
                              [-2.0357142857142865, 4.500000000000001, 0],
                              [-2.0535714285714293, 4.482142857142858, 0],
                              [-2.0357142857142865, 4.464285714285715, 0],
                              [-2.0178571428571432, 4.482142857142858, 0],
                              [-2.0357142857142865, 4.500000000000001, 0],
                            ],
                            [
                              null,
                              [-2.0357142857142865, 4.2857142857142865, 0],
                              [-2.0535714285714293, 4.267857142857144, 0],
                              [-2.0357142857142865, 4.250000000000001, 0],
                              [-2.0178571428571432, 4.267857142857144, 0],
                              [-2.0357142857142865, 4.2857142857142865, 0],
                            ],
                            [null],
                            [null],
                          ],
                        },
                      ],
                    },
                    {
                      type: 'sketch',
                      content: [
                        {
                          type: 'disjointAssembly',
                          content: [
                            {
                              type: 'disjointAssembly',
                              content: [
                                {
                                  type: 'paths',
                                  tags: ['color/red'],
                                  paths: [
                                    [
                                      [4.000000000000001, 4.000000000000001, 0],
                                      [
                                        -4.000000000000001,
                                        4.000000000000001,
                                        0,
                                      ],
                                      [-4.000000000000002, -4, 0],
                                      [4, -4.000000000000002, 0],
                                    ],
                                  ],
                                },
                              ],
                            },
                            {
                              type: 'paths',
                              tags: ['color/red'],
                              paths: [
                                [
                                  null,
                                  [-3.857142857142858, 4.625000000000001, 0],
                                  [-3.9107142857142865, 4.607142857142858, 0],
                                  [-3.9285714285714293, 4.571428571428572, 0],
                                  [-3.9285714285714293, 4.5357142857142865, 0],
                                  [-3.9107142857142865, 4.500000000000001, 0],
                                  [-3.875000000000001, 4.482142857142858, 0],
                                  [-3.8035714285714293, 4.464285714285715, 0],
                                  [-3.750000000000001, 4.446428571428572, 0],
                                  [-3.7142857142857153, 4.4107142857142865, 0],
                                  [-3.696428571428572, 4.375000000000001, 0],
                                  [-3.696428571428572, 4.321428571428572, 0],
                                  [-3.7142857142857153, 4.2857142857142865, 0],
                                  [-3.732142857142858, 4.267857142857144, 0],
                                  [-3.7857142857142865, 4.250000000000001, 0],
                                  [-3.857142857142858, 4.250000000000001, 0],
                                  [-3.9107142857142865, 4.267857142857144, 0],
                                  [-3.9285714285714293, 4.2857142857142865, 0],
                                  [-3.9464285714285725, 4.321428571428572, 0],
                                  [-3.9464285714285725, 4.375000000000001, 0],
                                  [-3.9285714285714293, 4.4107142857142865, 0],
                                  [-3.8928571428571437, 4.446428571428572, 0],
                                  [-3.8392857142857153, 4.464285714285715, 0],
                                  [-3.7678571428571437, 4.482142857142858, 0],
                                  [-3.732142857142858, 4.500000000000001, 0],
                                  [-3.7142857142857153, 4.5357142857142865, 0],
                                  [-3.7142857142857153, 4.571428571428572, 0],
                                  [-3.732142857142858, 4.607142857142858, 0],
                                  [-3.7857142857142865, 4.625000000000001, 0],
                                  [-3.857142857142858, 4.625000000000001, 0],
                                ],
                                [null],
                                [null],
                                [
                                  null,
                                  [-3.3035714285714293, 4.500000000000001, 0],
                                  [-3.1071428571428577, 4.250000000000001, 0],
                                ],
                                [
                                  null,
                                  [-3.1071428571428577, 4.500000000000001, 0],
                                  [-3.3035714285714293, 4.250000000000001, 0],
                                ],
                                [null],
                                [null],
                                [
                                  null,
                                  [-2.625000000000001, 4.625000000000001, 0],
                                  [-2.6785714285714293, 4.607142857142858, 0],
                                  [-2.696428571428572, 4.571428571428572, 0],
                                  [-2.696428571428572, 4.5357142857142865, 0],
                                  [-2.6785714285714293, 4.500000000000001, 0],
                                  [-2.6428571428571432, 4.482142857142858, 0],
                                  [-2.571428571428572, 4.464285714285715, 0],
                                  [-2.5178571428571432, 4.446428571428572, 0],
                                  [-2.4821428571428577, 4.4107142857142865, 0],
                                  [-2.464285714285715, 4.375000000000001, 0],
                                  [-2.464285714285715, 4.321428571428572, 0],
                                  [-2.4821428571428577, 4.2857142857142865, 0],
                                  [-2.500000000000001, 4.267857142857144, 0],
                                  [-2.5535714285714293, 4.250000000000001, 0],
                                  [-2.625000000000001, 4.250000000000001, 0],
                                  [-2.6785714285714293, 4.267857142857144, 0],
                                  [-2.696428571428572, 4.2857142857142865, 0],
                                  [-2.7142857142857153, 4.321428571428572, 0],
                                  [-2.7142857142857153, 4.375000000000001, 0],
                                  [-2.696428571428572, 4.4107142857142865, 0],
                                  [-2.6607142857142865, 4.446428571428572, 0],
                                  [-2.6071428571428577, 4.464285714285715, 0],
                                  [-2.5357142857142865, 4.482142857142858, 0],
                                  [-2.500000000000001, 4.500000000000001, 0],
                                  [-2.4821428571428577, 4.5357142857142865, 0],
                                  [-2.4821428571428577, 4.571428571428572, 0],
                                  [-2.500000000000001, 4.607142857142858, 0],
                                  [-2.5535714285714293, 4.625000000000001, 0],
                                  [-2.625000000000001, 4.625000000000001, 0],
                                ],
                                [null],
                                [null],
                                [
                                  null,
                                  [-2.0357142857142865, 4.500000000000001, 0],
                                  [-2.0535714285714293, 4.482142857142858, 0],
                                  [-2.0357142857142865, 4.464285714285715, 0],
                                  [-2.0178571428571432, 4.482142857142858, 0],
                                  [-2.0357142857142865, 4.500000000000001, 0],
                                ],
                                [
                                  null,
                                  [-2.0357142857142865, 4.2857142857142865, 0],
                                  [-2.0535714285714293, 4.267857142857144, 0],
                                  [-2.0357142857142865, 4.250000000000001, 0],
                                  [-2.0178571428571432, 4.267857142857144, 0],
                                  [-2.0357142857142865, 4.2857142857142865, 0],
                                ],
                                [null],
                                [null],
                              ],
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          },
        },
      ];

      const run = async () => {
        const body = document.getElementsByTagName('body')[0];
        const notebookElement = await toDomElement(notebook);
        body.appendChild(notebookElement);
      };

      document.onreadystatechange = () => {
        if (document.readyState === 'complete') {
          run();
        }
      };
    </script>
  </body>
</html>
